### PR TITLE
Added lite deleverageAndWithdraw function

### PR DESCRIPTION
**Pull request by thrilok209**

### CODE REVIEW

**Comment by pradyuman-verma** [comment]
### Deployed on mainnet

[0xD2f137b80D877115c48250034DFad618CAB95Fd3](https://etherscan.io/address/0xD2f137b80D877115c48250034DFad618CAB95Fd3)

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 1]
```diff
@@ -132,8 +138,70 @@ abstract contract InstaLiteConnector is Events, Basic {
...
+		uint256 deleverageAmount,
+		uint256 withdrawAmount,
+		uint256[] memory getIds,
+		uint256[] memory setIds
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		require(getIds.length > 2, "invalid get-ids length");
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
		require(getIds.length >= 2, "invalid get-ids length");
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 2]
```diff
@@ -132,8 +138,70 @@ abstract contract InstaLiteConnector is Events, Basic {
...
+			address(this)
+		);
+
+		uint256 _stETHAmt = astethToken.balanceOf(address(this)) -
+			initialBalStETH;
+		uint256 _tokenAmt = tokenContract.balanceOf(address(this)) -
+			initialBalToken;
+
+		// TODO: add require conditions
+		if (setIds.length > 2) {
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
		if (setIds.length >= 2) {
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 3]
```diff
@@ -132,8 +138,70 @@ abstract contract InstaLiteConnector is Events, Basic {
...
+
+		IInstaLite(vaultAddr).deleverageAndWithdraw(
+			_deleverageAmt,
+			_withdrawAmount,
+			address(this)
+		);
+
+		uint256 _stETHAmt = astethToken.balanceOf(address(this)) -
+			initialBalStETH;
+		uint256 _tokenAmt = tokenContract.balanceOf(address(this)) -
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]: If its eth vault, we get eth and steth, not weth

**Review by Vaibhav20201 — DISMISSED** [review]

**Comment by Vaibhav20201** [comment]
Deployed - [here](https://etherscan.io/address/0x3E5b9003e1674e15aA6a201b7B353C872C1ea791#code)

**Review by pradyuman-verma — APPROVED** [review]

**Pull request merged**